File: clojure/bob/src/bob.clj
(ns bob
    (:require [clojure.string :refer [upper-case ends-with? blank?]]))

(defn response-for [q]
      (cond
        (blank? q) "Fine. Be that way!"
        (and (re-find #"[a-zA-Z]" q) (= q (upper-case q)) ) "Whoa, chill out!"
        (ends-with? q "?") "Sure."
        :else "Whatever."))

File: elixir/bob/bob.exs
defmodule Bob do
  import String, only: [ends_with?: 2, upcase: 1, strip: 1]

  def hey(input) do
    cond do
        ends_with?(input, "?") -> "Sure."
        strip(input) == "" -> "Fine. Be that way!"
        (String.match?(input, ~r/\p{L}/) and upcase(input) == input) -> "Whoa, chill out!"
        true -> "Whatever."
    end
  end
end
